### PR TITLE
DISCUSSION - Enable model enforcement by default

### PR DESCRIPTION
**Pull request by vEpiphyte**
This PR enables model enforcement by default.  

This means that in order to add data to a cortex, the user must either use a built in model, or add their own data models to a Cortex.

### CODE REVIEW

File: synapse/cortex.py [hunk 1]
```diff
@@ -80,40 +83,68 @@ def openstore(url, storconf=None, **opts):
...
-        * postgres://[[<passwd>:]<user>@][<host>]/[<db>][/<table>]
+    Args:
+        url (str): URL which is parsed in order to connect to.
+        conf (dict): Configable options passed to the Cortex.
+        storconf (dict): Configable options passed to the storage layer.
+        **opts (dict): Additional options added to the link tufo from the
+            parsed URL.
+
+    Examples:
+        Open up  ram back cortex::
```
Comment (vertexmc) [inline synapse/cortex.py]: `backed`

File: synapse/tests/test_lib_storm.py [hunk 2]
```diff
@@ -334,7 +334,6 @@ def test_storm_tag_totag(self):
             node2 = core.formTufoByProp('inet:fqdn', 'vertex.vis')
             node3 = core.formTufoByProp('inet:url', 'https://vertex.link')
             node4 = core.formTufoByProp('inet:netuser', 'clowntown.link/pennywise')
-            node5 = core.formTufoByProp('geo:loc', 'derry')
```
Comment (vertexmc) [inline synapse/tests/test_lib_storm.py]: Why are we removing this line?

Comment (vEpiphyte) [inline synapse/tests/test_lib_storm.py]: wasn't used in the tests.

**Comment by vertexmc** [comment]
When do we plan on merging this PR? Are we waiting on anything specific?

**Comment by vEpiphyte** [comment]
@vertexmc our previous internal discussion was that we would leave this PR up for discussion for a few weeks; in order to allow people to test their code prior to the change.

We've only received one comment about it, privately, from a group which had no problems with the change.

**Comment by vEpiphyte** [comment]
Rebased ontop of master.

File: synapse/models/axon.py [hunk 3]
```diff
@@ -1,10 +1,40 @@
...
+        # with a default mode and nlinks value.
+        if ppfo:
+            return
+
+        pval, psubs = self.core.getTypeNorm('axon:path', pval)
+        # XXX Private static method should be promoted to a standalone method.
+        mode = self.core.getConfOpt('axon:dirmode')
+        pprops = s_axon.Axon._fs_new_dir_attrs(psubs.get('dir'), mode)
+        children = self.core.getTufosByProp('axon:path:dir', pval)
+        pprops['st_nlink'] += len(children) + 1 # XXX Is this correct?
```
Comment (vEpiphyte) [inline synapse/models/axon.py]: @vertexmc Is this correct for the st_nlink ?

File: synapse/axon.py [hunk 4]
```diff
@@ -898,7 +897,8 @@ def fs_mkdir(self, path, mode):
             dirn = self._getDirNode(ppath)
 
         attr = Axon._fs_new_dir_attrs(ppath, mode)
-        tufo = self.core.formTufoByProp('axon:path', path, **attr)
+        with self.flock():
```
Comment (vertexmc) [inline synapse/axon.py]: remove the `()` here

**Pull request merged**